src/edu/washington/escience/myria/expression/evaluate/GenericEvaluator.java

```diff
@@ -216,7 +217,7 @@ public EvaluatorResult evaluateColumn(final TupleBatch tb) throws InvocationTarg
         new TupleBuffer(Schema.ofFields(getExpression().getOutputName(), type));
     final WritableColumn resultsWriter = resultsBuffer.asWritableColumn(0);
     for (int rowIdx = 0; rowIdx < tb.numTuples(); ++rowIdx) {
-      eval(tb, rowIdx, countsWriter, resultsWriter, null);
+      eval(tb, rowIdx, tb, rowIdx, resultsWriter, countsWriter);
```

Comment:
This is somewhat a hack since this `tb` could be either the input tuple batch from child or a slice of the `TupleHashTable` state. However the stateful emit expressions only know to fetch data from the "state" even if the "state" serves more like an input tuple batch in this case. I added a comment and will think about how to make it more explicit.